src/hypothesis/database.py

```diff
@@ -240,7 +240,11 @@ def fetch(self, key):
     def save(self, key, value):
         path = self._value_path(key, value)
         if not os.path.exists(path):
-            tmpname = path + '.' + str(binascii.hexlify(os.urandom(16)))
+            suffix = binascii.hexlify(os.urandom(16))
+            if not isinstance(suffix, str):
```

Comment:
`binascii.hexlify` returns a (byte) `str` on Python 2, and `bytes` on Python 3.

Later (on line 247), a native string (i.e., bytes on Python 2, unicode on Python 3) is added to that. Therefore, it needs to be decoded if it's not a `str` (i.e. only on Python 3).